persistcache/persistcache.go

```diff
@@ -8,9 +8,11 @@ import (
 	"os"
 	"path/filepath"
 	"strings"
+	"sync"
 )
 
 type persistCache struct {
+	mu    sync.Mutex
```

Comment:
```suggestion
	    sync.Mutex
```
or do you expect more mutexes?

Reply:
not really, I'll add file mutex in case same cache will be referred from different processes (spoiler alert it will :D) peculiar syntax for me, so you don't name variable and then how do you exactly refer to it? I.e. what's alternative for

```
mu.Lock()
defer mu.Unlock()
```

Reply:
You use it like this:
```
type persistCache struct {
    root string
    sync.Mutex
}

func (pc *persistCache) doSomething() {
    pc.Lock()
    defer pc.Unlock()

    something(pc.root)
}
```

Reply:
Interesting mix of composition and inheritance... You live you learn, thanks @christoph-zededa !